src/framework/asset/asset.js

```diff
@@ -302,14 +312,19 @@ class Asset extends EventHandler {
         }
     }
 
+    /**
+     * Gets the file details or null if no file.
+     *
+     * @type {object}
```

Comment:
{object | null}

Reply:
There are literally hundreds of these kinds of corrections needed across the codebase. I'd rather leave fixing those out of this PR to simplify things for now (although you might see I have make one or two fixes as I went through documenting the getters). I'd say we just leave them to a follow up PR.